http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -0,0 +1,205 @@
+
+            case Some(resolvedCluster) => {
+
+              savedMasterMachineType match {
+                case Some(machineType) =>
+                  for {
+                    // perform gddao and db updates for new resources
+                    _ <- clusterHelper.updateMasterMachineType(resolvedCluster, MachineType(machineType))
+                    // start cluster
+                    _ <- clusterHelper.internalStartCluster(resolvedCluster)
+                    // clean-up info from follow-up table
+                    _ <- dbRef.inTransaction { followupQuery.delete(message.clusterFollowupDetails) }
+                  } yield ()
+                case None => IO.unit //the database has no record of a follow-up being needed. This is a no-op
+              }
+            }
+
+            case None => IO.raiseError(ClusterNotFoundException(message.clusterFollowupDetails.clusterId))
+          }
+        } yield result
+      }
+
+      //No actions for other statuses yet. There is some logic that will be needed for all other cases (i.e. the 'None' case where no cluster is found in the db and possibly the case that checks for the data in the DB)
+      // TODO: Refactor once there is more than one case
+      case _ => IO.unit
+    }
+
+}
+
+sealed trait LeoPubsubMessage {
+  def messageType: String
+}
+
+final case class StopUpdateMessage(updatedMachineConfig: MachineConfig, clusterId: Long) extends LeoPubsubMessage {
+  val messageType = "stopUpdate"
+}
+
+case class ClusterTransitionFinishedMessage(clusterFollowupDetails: ClusterFollowupDetails) extends LeoPubsubMessage {
+  val messageType = "transitionFinished"
+}
+
+final case class ClusterFollowupDetails(clusterId: Long, clusterStatus: ClusterStatus) extends Product with Serializable
+
+final case class PubsubException(message: String) extends Exception
+
+object LeoPubsubCodec {
+  implicit val stopUpdateMessageDecoder: Decoder[StopUpdateMessage] =
+    Decoder.forProduct2("updatedMachineConfig", "clusterId")(StopUpdateMessage.apply)
+
+  implicit val clusterFollowupDetailsDecoder: Decoder[ClusterFollowupDetails] =
+    Decoder.forProduct2("clusterId", "clusterStatus")(ClusterFollowupDetails.apply)
+
+  implicit val clusterTransitionFinishedDecoder: Decoder[ClusterTransitionFinishedMessage] =
+    Decoder.forProduct1("clusterFollowupDetails")(ClusterTransitionFinishedMessage.apply)
+
+  implicit val leoPubsubMessageDecoder: Decoder[LeoPubsubMessage] = Decoder.instance { message =>
+    for {
+      messageType <- message.downField("messageType").as[String]
+      value <- messageType match {
+        case "stopUpdate"         => message.as[StopUpdateMessage]
+        case "transitionFinished" => message.as[ClusterTransitionFinishedMessage]
+        case other                => Left(DecodingFailure(s"found a message with an unknown type when decoding: ${other}", List.empty))
+      }
+    } yield value
+  }
+
+  implicit val machineConfigEncoder: Encoder[MachineConfig] =
```

Comment:
can you move this encoder to `org.broadinstitute.dsde.workbench.leonardo.JsonCodec`